remoting/WFLY-19992-Enable-configuration-of-selected-XNIO-Remoting-parameters-for-remoting-objects-using-in-both-inbound-and-outbound-communication.adoc

```diff
@@ -0,0 +1,128 @@
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+ 
+* users would be able to configure selected options
+* users would be able to rely on one configuration file "standalone*.xml" for in server deployments
+* configuration available in standalone*.xml configuration - no additional configuration files are neccessary
+
+=== Non-Requirements
+
+* configuration of all available xnio/remoting options
+
+=== Future Work
+
+* extending the configuration so that different deployments could use diffent remoting object (see deployment plan)
+
+== Backwards Compatibility
+
+* if customers were relying on wildfly-config.xml that was provided to WildFly server for remoting configuration than it will cease to work (this was not the suggested solution though).
+
+=== Default Configuration
+
+* no changes
+
+=== Importing Existing Configuration
+
+* no need to import
+
+=== Deployments
+
+* no changes expected
+
+=== Interoperability
+
+* no changes expected
+
+== Implementation Plan
+
+I'm attaching relevant parts of my implementation plan that I wrote in WFCORE-6873:
+
+=== Current status
+There are number of XNIO objects created in default WildFly configuration (Endpoints, connectors) which have to be configured on both client and server side to make the configuration work. During the connection establishment, client and server negotiate the parameters, so if f.e. MAX_INBOUND_CHANNELS is set to high value on the server but MAX_OUTBOUND_CHANNELS is not explicitly set then the default value would still be used. Although, completely logical, it may be confusing to users who change configuration that the parameters they have modified are not used. Even with the knowledge above, it is not always clear what objects have to be changed to make the configuration work. For example, there is an endpoint configuration in the remoting subystem (not present in default subsystem configuration) that is meant to configure subsystem endpoint (that is the endpoint that is used by the subsystems and not management). The subystem endpoint is then used to create the http-connector that is used, among others, for remoting+http protocol invocations. The problem is that the properties configured at subsystem endpoint are not propagated to the http-connector. As a result, setting MAX_INBOUND_CHANNELS on the subsystem endpoint would not result in changing the number of channels, even if client side is configured correctly. In order to make server side work, the properties have to be configured on http-connector directly.
+
+There is no possibility to configure outbound endpoint in an elegant way. The outbound endpoint is obtained using Contextual API. By default, Endpoint provider looks for wildfly-config.xml file in the location provided by system variable and tries to parse this file. If no parameter is present then the empty endpoint is created.
```

Comment:
There are two ways to configure it: (1) modify the (many) relevant configuration files to accept general XNIO/Remoting properties, which involves a lot of changes and time but has the advantage of each deployment having control over its own XNIO/Remoting abstractions or (2) use system property overrides, as is aleady being done, to configure a small subset of properties that standalone application and deployment clients need to have control over, which is quick and easy to implement. Whether these are considered elegant is debatable.